### ISSUE TITLE
No sound is output

### ISSUE THREAD

**Issue by kmcoulson** [Alxandr/SpotiFire#66]
Hi, I cloned the repo for the first time today and it looks really good. However, I tried running the WinForms app to test it out and although everything seems to run fine no audio is output when a track is playing. Any idea why?

**Comment by Alxandr** [comment]
Sorry, no. I haven't looked at it in literally years, so I have no idea what might be causing it. One thing you can do though is to make sure that the music data is actually delivered to your sink by logging something or similar.

**Comment by NeoLegends** [comment]
> to make sure that the music data is actually delivered to your sink
This. libspotify is recently causing more and more problems itself. It's deprecated, after all.

**Comment by kmcoulson** [comment]
Thanks, but how do I check that? Where in the code?

**Comment by NeoLegends** [comment]
Insert a `Console.WriteLine` or any other hook of your choice inside [this method](https://github.com/Alxandr/SpotiFire/blob/fa5e39fb680b5d88e88df69092d42ee545cbdacd/SpotiFire.WinFormsTest/Form1.cs#L39).

**Comment by kmcoulson** [comment]
Oh, it is definitely hitting that event and the e.Samples length is not zero. However if I put a breakpoint [here](https://github.com/Alxandr/SpotiFire/blob/fa5e39fb680b5d88e88df69092d42ee545cbdacd/SpotiFire.TestClient/NAudioPlayer.cs#L28) the BufferedBytes is always zero (I don't know if that is right or not).

**Comment by NeoLegends** [comment]
Hah, of course. The samples are not added to the buffer. It just outputs "Enqueue" instead of enqueueing them. :D

**Comment by kmcoulson** [comment]
OK, could you give any advice on what needs to be done?

**Comment by kmcoulson** [comment]
I added this:
`buffer.AddSamples(samples, 0, samples.Length);`

before [this line](https://github.com/Alxandr/SpotiFire/blob/fa5e39fb680b5d88e88df69092d42ee545cbdacd/SpotiFire.TestClient/NAudioPlayer.cs#L31) and it seems to work. Does that seems like the correct thing to do?

**Comment by NeoLegends** [comment]
Exactly. I suggest you get a little into audio programming, ie. read about what PCM, audio samples, bit rates and sample rates are (and of course how libspotify's music delivery works). You'll get a much better understanding of whats happening within the code.

**Comment by kmcoulson** [comment]
Thanks, yeah I am reading up on it now. What I am trying to achieve is a server that shows up as a player in the spotify app so you can control what is playing from the native app. But then for it to broadcast whatever is playing and for that broadcast to be picked up and played by Raspberry Pis that are connected to speakers around the house. So I can play spotify in any room or multiple rooms.

**Comment by Alxandr** [comment]
Not doable unfortunately, until we get a new libspotify.
